### PR TITLE
made controlled map component

### PR DESCRIPTION
**Pull request by samrandhawa4**
added new options to make map component controlled: setZoom, setCentre
etc..

### CODE REVIEW

**Comment by ralphstodomingo** [comment]
Can anyone explain to me why this was merged?

The better thing to do was to separate the view prop into center and zoom, then update accordingly inside. Why the naming?

Upon looking at the source code and running examples, this breaks the draw interaction example. Making a circle there causes the map to zoom out. This doesn't show on the rawgit example because apparently that one isn't updated. Check the source of https://rawgit.com/allenhwkim/react-openlayers/master/app/index.html#/interactions/draw with dev tools.

File: src/map.tsx [hunk 1]
```diff
@@ -95,7 +100,17 @@ export class Map extends React.Component<any, any> {
...
   }
-
+  // update the view with new props
+  /* Modified by Harinder Randhawa */
+  componentWillReceiveProps(nextProps) {
+    let options = Util.getOptions(Object.assign(this.options, nextProps));
+    if(nextProps.setCenter !== this.props.view.center){
+      this.map.getView().setCenter(nextProps.setCenter);
+    }
+    if(nextProps.setZoom !== this.props.view.zoom){
```
Comment (ralphstodomingo) [inline src/map.tsx]: This throws errors while running the examples locally. The examples often don't have view defined as a prop. Do we require the view prop or not? If not, put a check in here.

File: app/interactions/draw.tsx [hunk 2]
```diff
@@ -13,25 +13,30 @@ var vectorSource = new ol.source.Vector({wrapX: false});
...
   }
-  
+
   render() {
     var typeSelect = document.getElementById('type');
-
     return (
       <div>
-        <Map view={{center: [-11000000, 4600000], zoom: 4}}>
+        <Map view={{center: [-11000000, 4600000], zoom: 4}} setCenter={ this.state.center }>
```
Comment (ralphstodomingo) [inline app/interactions/draw.tsx]: Am I the only one who sees how redundant this is? What keeps you from watching the view prop itself? And if you'll insist, why not remove the view prop, replace it with center and zoom, and update accordingly inside `componentWillReceiveProps`?

**Comment by ralphstodomingo** [comment]
I'll submit a PR to implement this better. Updating to the latest version broke our prototype map with this.

**Comment by allenhwkim** [comment]
@ralphstodomingov You seem to know ReactJS, please make another PR for it.

**Comment by ralphstodomingo** [comment]
I already have, @allenhwkim. I do advise we bump this as a major version so people will expect the changes, and the readme should be updated.

**Pull request merged**